### PR TITLE
Add ignoring bot accounts as Bot param

### PR DESCRIPTION
**Pull request by greenbigfrog**
Not sure if we should also add it as command param. (Also not sure, if this is how you would like it implemented, probably not...)

### CODE REVIEW

**Comment by meew0** [comment]
Still needs to be added to the CommandBot initializer so it can be used there.

File: lib/discordrb/bot.rb [hunk 1]
```diff
@@ -91,11 +91,12 @@ class Bot
     # @param num_shards [Integer] The total number of shards that should be running. See
     #   https://github.com/hammerandchisel/discord-api-docs/issues/17 for how to do sharding.
     # @param redact_token [true, false] Whether the bot should redact the token in logs. Default is true.
+    # @param ignore_bots [true, false] Whether the bot should ignore Bot accounts or not. Default is false.
```
Comment (meew0) [inline lib/discordrb/bot.rb]: "ignore bot accounts"

File: lib/discordrb/commands/command_bot.rb [hunk 2]
```diff
@@ -66,6 +66,7 @@ class CommandBot < Discordrb::Bot
     #   :advanced_functionality). Default is '"'.
     # @option attributes [String] :quote_end Character that should end a quoted string (see
     #   :advanced_functionality). Default is '"'.
+    # @option attributes [true, false] :ignore_bots Whether the bot should ignore Bot accounts or not. Default is false.
```
Comment (meew0) [inline lib/discordrb/commands/command_bot.rb]: "ignore bot accounts"

File: lib/discordrb/bot.rb [hunk 3]
```diff
@@ -917,6 +919,11 @@ def handle_dispatch(type, data)
           return
         end
 
+        if @ignore_bots && user(data['author']['id']).bot_account?
```
Comment (meew0) [inline lib/discordrb/bot.rb]: Why does there need to be a resolution here? The author's data should already contain whether it is a bot or not.

File: lib/discordrb/bot.rb [hunk 4]
```diff
@@ -917,6 +919,11 @@ def handle_dispatch(type, data)
           return
         end
 
+        if @ignore_bots && user(data['author']['bot'])
```
Comment (meew0) [inline lib/discordrb/bot.rb]: Did you even test this? `bot` isn't an ID.

Comment (greenbigfrog) [inline lib/discordrb/bot.rb]: lol...
yes, I did "test" it... But I know what went wrong when testing... was with my mind in crystal/js and fergot to build and install the gem
my bad

**Pull request merged**